include/fiction/algorithms/simulation/sidb/operational_domain.hpp

```diff
@@ -490,6 +609,83 @@ class operational_domain_impl
 
         return op_domain;
     }
+    /**
+     * Performs a grid search over the specified parameter ranges. For each physical
+     * parameter combination found for which the given CDS is physically valid, it is determined whether the CDS is the
+     * ground state or the n-th excited state.
+     *
+     * @param lyt SiDB cell-level layout that is simulated and compared to the given CDS.
+     * @return All physically valid physical parameters and the excited state number.
+     */
+    [[nodiscard]] operational_domain<parameter_point, uint64_t>
+    grid_search_for_physically_valid_parameters(Lyt& lyt) noexcept
```

Comment:
There are still some lines of code in this function that are not covered by any tests yet. Do you think, you think you could add some respective test cases? Otherwise, this PR looks good to be merged :pray:

Reply:
Sure! I'll have a look!